File: app/src/main/java/com/example/dropex/adpaters/OrderGVAdapter.java
package com.example.dropex.adpaters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.dropex.R;
import com.example.dropex.models.OrdersModel;

import java.util.ArrayList;

public class OrderGVAdapter extends ArrayAdapter<OrdersModel> {

    public OrderGVAdapter(@NonNull Context context, ArrayList<OrdersModel> courseModelArrayList) {
        super(context, 0, courseModelArrayList);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        View listItemView = convertView;
        if (listItemView == null) {
            // Layout Inflater inflates each item to be displayed in GridView.
            listItemView = LayoutInflater.from(getContext()).inflate(R.layout.order_item, parent, false);
        }

        OrdersModel ordersModel = getItem(position);
        TextView productTVName = listItemView.findViewById(R.id.productNameTV);
        ImageView productImageIV = listItemView.findViewById(R.id.productImageIV);
        TextView quantityTV = listItemView.findViewById(R.id.quantityTV);
        TextView sizeTV = listItemView.findViewById(R.id.sizeTV);
        TextView totalPriceTV = listItemView.findViewById(R.id.totalPriceTV);

        String sizeText = "Size: " + ordersModel.getSize().toString().toUpperCase();
        String quantityText = "Quantity: " + ordersModel.getQuantity();

        productTVName.setText(R.string.placeholder);
        productImageIV.setImageResource(R.drawable.sample);
        quantityTV.setText(quantityText);
        sizeTV.setText(sizeText);
        totalPriceTV.setText("$3232");
        return listItemView;
    }
}

File: app/src/main/java/com/example/dropex/adpaters/OrderGVAdapter.kt
package com.example.dropex.adpaters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import coil.load
import com.example.dropex.R
import com.example.dropex.models.OrdersModel
import com.example.dropex.models.ProductModel
import java.util.Locale

class OrderGVAdapter(context: Context, ordersModelArrayList: ArrayList<OrdersModel?>, private var productModelArrayList: ArrayList<ProductModel?>) : ArrayAdapter<OrdersModel?>(context, 0, ordersModelArrayList) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var listItemView = convertView
        if (listItemView == null) {
            listItemView = LayoutInflater.from(context).inflate(R.layout.order_item, parent, false)
        }

        val orderModel = getItem(position)
        val productModel = productModelArrayList[position]
        val productTVName = listItemView!!.findViewById<TextView>(R.id.productNameTV)
        val productImageIV = listItemView.findViewById<ImageView>(R.id.productImageIV)
        val quantityTV = listItemView.findViewById<TextView>(R.id.quantityTV)
        val sizeTV = listItemView.findViewById<TextView>(R.id.sizeTV)
        val totalPriceTV = listItemView.findViewById<TextView>(R.id.totalPriceTV)

        orderModel!!
        productModel!!

        val priceText = "$" + orderModel.quantity * productModel.price
        val sizeText = "Size: " + orderModel.size.uppercase(Locale.getDefault())
        val quantityText = "Quantity: " + orderModel.quantity.toString().uppercase(Locale.getDefault())

        productTVName.text = productModel.name
        productImageIV.load(productModel.imgUrl)
        totalPriceTV.text = priceText
        quantityTV.text = quantityText
        sizeTV.text = sizeText
        return listItemView
    }
}